core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -672,9 +675,28 @@ private void simplifyAndTerms(List<RexNode> terms, RexUnknownAs unknownAs) {
     RexSimplify simplify = this;
     for (int i = 0; i < terms.size(); i++) {
       RexNode t = terms.get(i);
-      if (Predicate.of(t) == null) {
+      Predicate predicate = Predicate.of(t); 
+      if (predicate == null) {
         continue;
       }
+      
+      //add for array-type
```

Comment:
Better to add a blank to `//` and `add` for line comment.

Reply:
yeah, thank you!